test/functional/wallet_importmulti.py

```diff
@@ -32,595 +80,523 @@ def skip_test_if_missing_module(self):
 
-    def run_test (self):
+    def get_key(self):
+        """Generate a fresh key on node0
+
+        Returns a named tuple of privkey, pubkey and all address and scripts."""
+        addr = self.nodes[0].getnewaddress()
+        pubkey = self.nodes[0].getaddressinfo(addr)['pubkey']
+        pkh = hash160(hex_str_to_bytes(pubkey))
+        return Key(self.nodes[0].dumpprivkey(addr),
+                   pubkey,
+                   CScript([OP_DUP, OP_HASH160, pkh, OP_EQUALVERIFY, OP_CHECKSIG]).hex(),  # p2pkh
+                   key_to_p2pkh(pubkey),  # p2pkh addr
+                   CScript([OP_0, pkh]).hex(),  # p2wpkh
+                   key_to_p2wpkh(pubkey),  # p2wpkh addr
+                   CScript([OP_HASH160, hash160(CScript([OP_0, pkh])), OP_EQUAL]).hex(),  # p2sh-p2wpkh
+                   CScript([OP_0, pkh]).hex(),  # p2sh-p2wpkh redeem script
+                   key_to_p2sh_p2wpkh(pubkey))  # p2sh-p2wpkh addr
+
+    def get_multisig(self):
+        """Generate a fresh multisig on node0
+
+        Returns a named tuple of privkeys, pubkeys and all address and scripts."""
+        addrs = []
+        pubkeys = []
+        for i in range(3):
+            addr = self.nodes[0].getaddressinfo(self.nodes[0].getnewaddress())
+            addrs.append(addr['address'])
+            pubkeys.append(addr['pubkey'])
+        script_code = CScript([OP_2] + [hex_str_to_bytes(pubkey) for pubkey in pubkeys] + [OP_3, OP_CHECKMULTISIG])
+        witness_script = CScript([OP_0, sha256(script_code)])
+        return Multisig([self.nodes[0].dumpprivkey(addr) for addr in addrs],
+                        pubkeys,
+                        CScript([OP_HASH160, hash160(script_code), OP_EQUAL]).hex(),  # p2sh
+                        script_to_p2sh(script_code),  # p2sh addr
+                        script_code.hex(),  # redeem script
+                        witness_script.hex(),  # p2wsh
+                        script_to_p2wsh(script_code),  # p2wsh addr
+                        CScript([OP_HASH160, witness_script, OP_EQUAL]).hex(),  # p2sh-p2wsh
+                        script_to_p2sh_p2wsh(script_code))  # p2sh-p2wsh addr
+
+    def test_importmulti(self, req, success, error_code=None, error_message=None):
+        """Run importmulti and assert success"""
+        result = self.nodes[1].importmulti([req])
+        assert_equal(result[0]['success'], success)
+        if error_code is not None:
+            assert_equal(result[0]['error']['code'], error_code)
+            assert_equal(result[0]['error']['message'], error_message)
+
+    def test_address(self, address, **kwargs):
+        """Get address info for `address` and test whether the returned values are as expected."""
+        addr_info = self.nodes[1].getaddressinfo(address)
+        for key, value in kwargs.items():
+            if addr_info[key] != value:
+                raise(AssertionError("key {} value {} did not match expected value {}".format(key, addr_info[key], value)))
```

Comment:
Nit: `raise AssertionError(…)` is more idiomatic and consistent with the rest of the code base. Drop the unnecessary parens :-)